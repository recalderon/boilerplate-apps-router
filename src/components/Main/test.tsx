import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('renderiza o heading', () => {
    // Renderiza o componente
    const {container} = render(<Main />)
    // Busca o elemento e verifica se ele existe
    expect(screen.getByRole('heading', { name: /react avan√ßado/i })).toBeInTheDocument()
    // Gerar snapshot, garante a estrutura do componente
    expect(container.firstChild).toMatchSnapshot()
  })
})
