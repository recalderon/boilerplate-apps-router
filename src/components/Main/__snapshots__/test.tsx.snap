// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renderiza o heading 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
